stream_alert/alert_processor/output_base.py

```diff
@@ -0,0 +1,319 @@
+        temp_dir = os.path.join(tempfile.gettempdir(), "stream_alert_secrets")
+
+        # Check if this item exists as a file, and remove it if it does
+        if os.path.exists(temp_dir) and not os.path.isdir(temp_dir):
+            os.remove(temp_dir)
+
+        # Create the folder on disk to store the credentials temporarily
+        if not os.path.exists(temp_dir):
+            os.makedirs(temp_dir)
+
+        return temp_dir
+
+    def _load_creds(self, descriptor):
+        """First try to load the credentials from /tmp and then resort to pulling
+        the credentials from S3 if they are not cached locally
+
+        Args:
+            descriptor [string]: unique identifier used to look up these credentials
+
+        Returns:
+            [dict] the loaded credential info needed for sending alerts to this service
+        """
+        local_cred_location = os.path.join(self._local_temp_dir(),
+                                           self.output_cred_name(descriptor))
+
+        # Creds are not cached locally, so get the encrypted blob from s3
+        if not os.path.exists(local_cred_location):
+            if not self._get_creds_from_s3(local_cred_location, descriptor):
+                return
+
+        with open(local_cred_location, 'wb') as cred_file:
+            enc_creds = cred_file.read()
+
+        # Get the decrypted credential json from kms and load into dict
+        # This could be None if the kms decryption fails, so check it
+        decrypted_creds = self._kms_decrypt(enc_creds)
+        if not decrypted_creds:
+            return
+
+        creds_dict = json.loads(decrypted_creds)
+
+        # Add any of the hard-coded default output props to this dict (ie: url)
+        defaults = self._get_default_properties()
+        if defaults:
+            creds_dict.update(defaults)
+
+        return creds_dict
+
+    def _format_s3_bucket(self, suffix):
+        """Format the s3 bucket by combining the stored qualifier with a suffix
+
+        Args:
+            suffix [string]: Suffix for an s3 bucket
+
+        Returns:
+            [string] The combined prefix and suffix
+        """
+        return '.'.join([self.s3_prefix, suffix])
+
+    def _get_creds_from_s3(self, cred_location, descriptor):
+        """Pull the encrypted credential blob for this service and destination from s3
+
+        Args:
+            cred_location [string]: The tmp path on disk to to store the encrypted blob
+            descriptor [string]: Service destination (ie: slack channel, pd integration)
+
+        Returns:
+            [boolean] True if download of creds from s3 was a success
+        """
+        try:
+            client = boto3.client('s3', region_name=self.region)
+            with open(cred_location, 'wb') as cred_output:
+                client.download_fileobj(self.get_secrets_bucket_name(),
+                                        self.output_cred_name(descriptor),
+                                        cred_output)
```

Comment:
Why not just [`client.download_file`](http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.download_file)? Then you don't need the file object.